### ISSUE TITLE
When train the model with custom data, label must be same?

### ISSUE THREAD

**Issue by studyfirston** [ultralytics/ultralytics#2671]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello.
When I train the model using my own data, I wonder I have to match label number?

example
ultralytics/datasets/coco.yaml
In the coco.yaml, the person are designated as zero.

If my data has 3 class, (apple, banana, person) and I train 0 : apple, 1 : banana, 2 : person
then, does YOLO pretrained model continue to learn about person?

**Comment by glenn-jocher** [comment]
@studyfirston hello,

Yes, you must match the label numbers between your custom data and the pre-trained model. In your example, since "person" in the pre-trained model is labeled as 0 in coco.yaml, you would need to label "person" as 0 in your custom data as well. When you train your custom data with YOLOv8, the model will continue to learn about all three classes in your data, including "person". 

I hope this helps! Let us know if you have any other questions.

**Comment by studyfirston** [comment]
@glenn-jocher 

Oh my god, really?
I just thought if there is same label name between my data and pre-trained data, then model can match the label between them.
Because pre-trained model using my data already find object well.

When I think about it after listening to you, my model will be better if I match label number about person.
I need to change the labelling file by designating the person as zero.

Thank you for helping me. Your answer is so helpful for me. 
I wish you all the best!

**Comment by glenn-jocher** [comment]
@studyfirston, I'm glad that I could be of help to you. It is true that labeling your custom data with the same label numbers as the pre-trained model will help the model learn better. The pre-trained model has already learned features that are generalizable to many types of objects, but those features are associated with specific label numbers. If you match your custom data's labels to those label numbers, then the model can more easily transfer what it learned from the pre-trained model to your specific data. Good luck with your training, and please feel free to reach out if you have any more questions!
